src/main/scala/com/zeotap/data/io/source/spark/constructs/DataFrameOps.scala

```diff
@@ -71,6 +76,52 @@ object DataFrameOps {
 
       dataFrame.withColumn("inputPathColumn", addInputPathColumn(input_file_name())).select("inputPathColumn").distinct().collect.map(row => row.getString(0))
     }
+
+    /**
+     * Partitions given dataframe into number of partitions provided and writes the
+     * partitioned dataframe to the intermediate path provided, so that parallel processing can take place.
+     *
+     * @param numberOfPartitions         is the number of partitions we want in the dataframe.
+     * @param intermediatePath           is the intermediate path in which the partitioned data frame is written.
+     * @param prioritiseIntermediatePath is boolean value which denotes whether the intermediate path (i.e already partitioned path)
+     *                                   needs to prioritised or should we force repartition. If it is true then data at intermediate path
+     *                                   will be returned(if non empty).
+     * @return Returns Dataframe with specified number of partitions.
+     */
+
+    def split(numberOfPartitions: Int, intermediatePath: String, prioritiseIntermediatePath: Boolean)(implicit spark: SparkSession): DataFrame = {
```

Comment:
create two more versions of split 
1. where you don't take the `numberOfPartitions` and uses default 200
2. where you don't take the `numberOfPartitions` and `prioritiseIntermediatePath` and uses default 200 and true

Reply:
Don't pass spark session as an implicit argument here, instead use dataframe.sparkSession

Reply:
addressed